cogs/utils/generic.py

```diff
@@ -60,9 +65,29 @@ async def get_channels(self, ctx):
         await ctx.channel.send(f"```{output}```")
 
 
-    # Do stuff when a message is deleted
-    @commands.Cog.listener()
-    async def on_message_delete(self, message):
-        pass
+    # Send a message to a user on their birth day
+    @tasks.loop(hours=24)
+    async def birthday_alert(self):
+        # List to save id of members who already recieved a msg
+        member_ids_sent = []
+        guild_ids = await get_all_guilds()
+        for guild_id in guild_ids:
+            members = await get_all_members(guild_id)
+            for member in members:
+                member_id = member['member_id']
+                if member['birthday'] == None or member_id in member_ids_sent:
+                    continue
+                bday = datetime.strptime(member['birthday'], '%Y-%m-%d')
+                now = date.today()
+                if (bday.month, bday.day) == (now.month, now.day):
+                    user = self.bot.get_user(member_id)
+                    reply = f"Happy birthday, **{user.name}**!"
+                    await user.send(content=reply)
+                    member_ids_sent.append(member_id)
```

Comment:
This is great! The only thing I might change here would be to simplify the process by adding a new command to db.py that simply filters the member table by those that have a birthday set, and then returns that list for iterating (i.e. some command called `get_member_birthdays()`. This avoids needing to loop over guilds and every single member in a guild, and SQL filtering is extremely efficient and low resource usage, and thus would scale better in the long term.